Dataset Collector #1, Shhanukova Milana - 19FPL1

MilanaShhanukova opened the pull request:
Changed the score and added class Crawler the first version

Commit:
Update scrapper.py

Commit:
Update target_score.txt

Commit:
Update constants.py

Commit:
Update crawler_config.json

Commit:
Update scrapper.py

Commit:
Update scrapper.py

Commit:
Update scrapper.py

Commit:
Update scrapper.py

Commit:
Update scrapper.py

Commit:
Update article.py

Commit:
Update constants.py

Commit:
Update constants.py

Commit:
Update article.py

Commit:
Update scrapper.py

Commit:
Update constants.py

Commit:
Update requirements.txt

Commit:
changed configuration

demid5111 commented on constants.py:
```diff
@@ -7,3 +7,5 @@
 PROJECT_ROOT = os.path.dirname(os.path.realpath(__file__))
 ASSETS_PATH = os.path.join(PROJECT_ROOT, 'tmp', 'articles')
 CRAWLER_CONFIG_PATH = os.path.join(PROJECT_ROOT, 'crawler_config.json')
+HEADERS = {"User-Agent": "Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) \
```
headers are duplicated in constants and in configuration file

demid5111 commented on article.py:
```diff
@@ -46,7 +46,6 @@ def save_raw(self):
                       indent=4,
                       ensure_ascii=False,
                       separators=(',', ': '))
-    
```
discard changes

demid5111 commented on requirements.txt:
```diff
@@ -0,0 +1,3 @@
+requests==2.25.1
```
alphabetical order

demid5111 commented on scrapper.py:
```diff
@@ -1,6 +1,15 @@
 """
 Crawler implementation
 """
+import json
```
1. imports in alphabetical order
2. imports in following order: Python standard library, empty line, external libraries, empty line, local modules

demid5111 commented on scrapper.py:
```diff
@@ -27,71 +37,159 @@ class UnknownConfigError(Exception):
     """
-    def __init__(self, seed_urls: list, max_articles: int):
-        pass
+    def __init__(self, main_urls, num_articles, max_articles, headers):
```
changing of interface is not allowed

demid5111 commented on scrapper.py:
```diff
@@ -27,71 +37,159 @@ class UnknownConfigError(Exception):
+    def _extract_url(article_bs, headers):
+        urls = []
+
+        req = requests.get(article_bs, headers)
```
_extract_url should not be calling the page, it should just extract

demid5111 commented on scrapper.py:
```diff
@@ -27,71 +37,159 @@ class UnknownConfigError(Exception):
     def _fill_article_with_text(self, article_soup):
-        pass
+        text_from_soup = article_soup.find(class_="video-show").find_all("p")
+        clean_text = [str(sent)[3:-4] for sent in text_from_soup]
```
what are these magic constants about - 3 and 4?

demid5111 commented on scrapper.py:
```diff
@@ -27,71 +37,159 @@ class UnknownConfigError(Exception):
+        }
+        date = date_str.split()
+        date[1] = ru_months[date[1]]
+        return date
```
it is not the datetime format (!)

demid5111 commented on scrapper.py:
```diff
@@ -27,71 +37,159 @@ class UnknownConfigError(Exception):
+        raise UnknownConfigError
+
+    if not isinstance(conf['base_urls'], list) or \
+            not all([isinstance(seed_url, str) for seed_url in conf['base_urls']]):
```
there should be a better check for urls

Commit:
maybe it's what you wanted....

Commit:
order

Commit:
deleted headers from config

Commit:
some updates

Commit:
please

Commit:
Update scrapper.py

Commit:
Update article.py

Commit:
Update article.py

Commit:
are kidding me?

Commit:
no whitespaces!!

dmitry-uraev committed:
Merge remote-tracking branch 'upstream/main' into HEAD

dmitry-uraev committed:
Merge remote-tracking branch 'upstream/main' into HEAD

demid5111 commented:
I see no activity in your PR about pipeline implementation. The risk to miss the coming final deadline (April, 5th) is growing. Please secure you make implementation in time.

Commit:
the first version

It is not that obvious for me about checks for config_path, so please tell me, am i on the right way or not?

Commit:
little updates with path

Commit:
some pos building

Commit:
changes for pylint

Commit:
let's try :)

Commit:
Update requirements.txt

demid5111 commented on pipeline.py:
```diff
@@ -16,74 +20,103 @@ class InconsistentDatasetError(Exception):
         Register each dataset entry
         """
-        pass
+        files_txt = [str(file.name) for file in Path(self.path_to_raw_txt_data).glob('**/*.txt')]
```
we should not go from Path abstraction to string again.

MilanaShhanukova replied:
it is the way to get the index of  file. As it is said in documentation it is appropriate to do.

demid5111 commented on pipeline.py:
```diff
@@ -16,74 +20,103 @@ class InconsistentDatasetError(Exception):
         Returns storage params
         """
-        pass
+        self._scan_dataset()
```
read the requirements on where we nee to call scanning

demid5111 commented on pipeline.py:
```diff
@@ -16,74 +20,103 @@ class InconsistentDatasetError(Exception):
+        articles = self.corpus_manager.get_articles()  # dict with key - num, value - article instance
+        for article in articles.values():
+            raw_text = article.get_raw_text()
+            tokens = [i.__str__() for i in  self._process(raw_text)]
```
you are one of top students in the course. do you really mean calling __str__ method like this?

demid5111 commented on pipeline.py:
```diff
@@ -16,74 +20,103 @@ class InconsistentDatasetError(Exception):
+            raw_text = article.get_raw_text()
+            tokens = [i.__str__() for i in  self._process(raw_text)]
+            article.save_processed(" ".join(tokens))
+    @classmethod
```
you are not allowed to change the API - `_process` method accepts **instance** and not the **class** object

MilanaShhanukova replied:
it is the requirement of pylint, I'm not allowed to change the function.

demid5111 commented on pipeline.py:
```diff
@@ -16,74 +20,103 @@ class InconsistentDatasetError(Exception):
+        morphological_tokens = []
+        for word in raw_tokens:
+            if word["text"].isalpha() and word["analysis"]:
+                morph_token = MorphologicalToken(word["text"], word["analysis"][0]["lex"])
```
you should check for lext and gr as well (!)

MilanaShhanukova replied:
If there is analysis of the word, it means that all of the parametrs are there.

demid5111 commented on pipeline.py:
```diff
@@ -16,74 +20,103 @@ class InconsistentDatasetError(Exception):
+            if word["text"].isalpha() and word["analysis"]:
+                morph_token = MorphologicalToken(word["text"], word["analysis"][0]["lex"])
+                morph_token.mystem_tags = word["analysis"][0]["gr"]
+                morph_token.pymorthy_tags = str(MorphAnalyzer().parse(word["text"])[0].tag)
```
the same

demid5111 commented on pipeline.py:
```diff
@@ -16,74 +20,103 @@ class InconsistentDatasetError(Exception):
+        raise FileNotFoundError
+    if not path.is_dir():
+        raise NotADirectoryError
+    checks = list(path.glob('**/*.txt'))
```
there should be a check for the IDs - read requirements

demid5111 commented on pos_frequency_pipeline.py:
```diff
@@ -1,8 +1,32 @@
 Implementation of POSFrequencyPipeline for score ten only.
 """
-
+import collections
```
6 files when we requested 5

demid5111 commented on pos_frequency_pipeline.py:
```diff
@@ -1,8 +1,32 @@
-        pass
+    def run(self):
+        # we guess that all files are already preprocessed in a dir
+        pattern = re.compile("\(\w+")
```
what are the tags you are trying to get from the files?

MilanaShhanukova replied:
Tags from pymorphy can be found in such pattern.

demid5111 commented on pos_frequency_pipeline.py:
```diff
@@ -1,8 +1,32 @@
+        # we guess that all files are already preprocessed in a dir
+        pattern = re.compile("\(\w+")
+        path = Path(ASSETS_PATH)
+        files = list(path.glob('**/*.txt'))
```
you need to work on top of processed files not all txt files (!)

demid5111 commented on pos_frequency_pipeline.py:
```diff
@@ -1,8 +1,32 @@
 """
-
+import collections
+from pathlib import Path
```
pre-processing is hanging

demid5111 reviewed (changes requested):
need more work

Commit:
Update pipeline.py

Commit:
Update pos_frequency_pipeline.py

demid5111 commented on requirements.txt:
```diff
@@ -0,0 +1,7 @@
+lxml==4.6.2
+requests==2.25.1
+matplotlib==3.4.1
+numpy==1.20.2
```
why do you need numpy?

demid5111 commented on pos_frequency_pipeline.py:
```diff
@@ -1,8 +1,32 @@
+        path = Path(ASSETS_PATH)
+        files = list(path.glob('**/*_processed.txt'))
+        if not files:
+            print("no info to plot")
```
forgot

demid5111 reviewed (changes requested):
more work

Commit:
deleted classmethod

Commit:
some mistake about the number

Commit:
Update pipeline.py

Commit:
a small bug

Commit:
Merge branch 'main' of https://github.com/MilanaShhanukova/2020-2-level-ctlr into main

Commit:
Update scrapper.py

demid5111 commented:
Web site is not working. Based on historical data - I accept this work and give 10

Commit:
-

MilanaShhanukova commented:
Now bragazeta saw that the deadline is off and they decided to make their link work again. So green color, thanks.
